### PR TITLE
fix: Fixes description indenting

### PR DESCRIPTION
**Pull request by ctasada**
When the description field includes a long description it may not be properly rendered, mainly in Kotlin, where the way of managing the text-blocks is different.

 This PR improves the way descriptions are managed by:

 - Removes empty lines at the beginning of the text-block
 - Removes indention due to alignment, but keeps indention in the lines that need it

### CODE REVIEW

**Comment by ctasada** [comment]
I think this issue may be exclusive from Kotlin, when using text-blocks. The change should be transparent (or nearly transparent) for Java code, but we can wrap it in a special setting if needed.

**Comment by timonback** [comment]
I am assuming you use multi–line strings aka text blocks.
If it is kotlin specific, would it be sufficient for the users to call trimIndent themselves?

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/trim-indent.html

(I haven't looked through the code. This sounds like a method that should be part of a standard lib)

**Comment by ctasada** [comment]
Sadly it's not possible. Annotations must be compile-time constants, so it's not possible to do any manipulation, like `.trimIndent()`

**Comment by timonback** [comment]
Argh. Unfortunate that https://youtrack.jetbrains.com/issue/KT-22505/Support-calls-to-trimMargin-trimIndent-as-initializer-for-const-vals is not getting resolved.
And there seems to be no easy Apache Commons or similar implementation.

Will review the PR on Friday

File: springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    private static int resolveMinIndent(String[] lines) {
+        return Arrays.stream(lines)
+                .filter(line -> !line.trim().isEmpty())
+                .mapToInt(DescriptionUtil::countLeadingSpaces)
+                .min()
+                .orElse(0);
+    }
+
+    private static int countLeadingSpaces(String line) {
+        for (int i = 0; i < line.length(); i++) {
```
Comment (sam0r040) [inline springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java:76]: I think we could use StringUtils.indexOfAnyBut(line, ' ') instead

File: springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /**
+     * @return {@code true} if the line ends with a {@code '\n'} or {@code false} otherwise
+     */
+    private static boolean endsWithNewline(String text) {
+        if (text == null || text.isEmpty()) {
+            return false;
+        }
+        return text.charAt(text.length() - 1) == '\n';
```
Comment (sam0r040) [inline springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java:64]: We should be able to use `StringUtils.endsWith(text, "\n")` instead

File: springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        final String newLine = System.lineSeparator();
+        String[] lines = text.split(newLine);
+        int firstNonEmptyLine = findFirstNonEmptyIndex(lines);
+        if (firstNonEmptyLine != -1) {
+            lines = Arrays.copyOfRange(lines, firstNonEmptyLine, lines.length);
+        }
+        int minIndent = resolveMinIndent(lines);
+        var result = Arrays.stream(lines)
+                .map(line -> line.substring(Math.min(line.length(), minIndent)))
```
Comment (sam0r040) [inline springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java:34]: Could we simplify by using `StringUtils.substring(line, minIndent)`?

**Comment by sam0r040** [comment]
Kind of unfortunate that there is no nice solution and we have to provide the Utils for it. I think I found some simplifications, but not as much as I wished :(. However, I would add two more test cases: String with no empty lines and null input.

File: springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+package io.github.springwolf.core.asyncapi.scanners.common.utils;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+
+/**
+ * Auxiliary class to manage Description or long texts.
+ */
+public class DescriptionUtil {
```
Comment (sam0r040) [inline springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/utils/DescriptionUtil.java:11]: Have you considered naming it TextUtil? I think the functionality is independent of descriptions.

**Comment by ctasada** [comment]
@sam0r040 All the comments addressed. I also added more tests covering the different corner cases. Coverage is now 100%

**Review by timonback — APPROVED** [review]
